yktrace/src/tir.rs

```diff
@@ -597,7 +544,7 @@ pub struct Guard {
     /// The TIR locals that are live at the time of the guard (in addition to the trace I/O
     /// variable, $1, which is assumed to be live throughout the entirety of the trace). This is
     /// needed so that we can initialise the blackhole interpreter with the correct state.
-    pub live_locals: Vec<Local>
+    pub live_locals: Vec<(Local, Local)>
```

Comment:
Why a pair?

Reply:
It's a mapping from a TIR local to its SIR equivalent. When we blackhole we need to copy the value from the TIR local to a SIR local. Due to renaming when converting from SIR to TIR, they don't always share the same name.

Reply:
OK, let's document that clearly (the doc comment feels a bit out of date?).

Reply:
Or make a struct with named fields?

Reply:
> Or make a struct with named fields?

Good idea: it'll make it harder to mix the two `Locals` up.

Reply:
Fixed in 8ce69fe

Reply:
Perhaps we raced? Were we expecting a new struct? I'm not *too* fussed if not.

Reply:
Fixing.

Reply:
Fixed in f60d82c.